File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/csharp/src/de/sots/cellarsandcentaurs/adapter/persistence/PersistedCreatures.cs
using de.sots.cellarsandcentaurs.domain.model;
using de.sots.cellarsandcentaurs.domain.service;
using System;
using Microsoft.Extensions.DependencyInjection;

namespace de.sots.cellarsandcentaurs.adapter.persistence
{
    [Service]
    public class PersistedCreatures : Creatures
    {
        private readonly CreatureRepository repository;

        public PersistedCreatures(CreatureRepository repository)
        {
            this.repository = repository;
        }

        public void Save(Creature creature)
        {
            repository.Add(new CreatureEntity(creature.Id.Id));
        }

        public Creature Find(CreatureId id)
        {
            var creatureEntity = repository.Find(id.Id) ?? throw new NoSuchCreatureException(id);
            return new Creature(new CreatureId(creatureEntity.Id));
        }
    }
}

File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/cpp/src/de/sots/cellarsandcentaurs/adapter/persistence/PersistedCreatures.cpp
#include "PersistedCreatures.h"
#include "../../domain/model/Creature.h"
#include "../../domain/model/NoSuchCreatureException.h"

namespace de::sots::cellarsandcentaurs::adapter::persistence {

PersistedCreatures::PersistedCreatures(std::shared_ptr<CreatureRepository> repository)
    : repository_{std::move(repository)} {}

void PersistedCreatures::save(const domain::model::Creature& creature) {
    repository_->save(CreatureEntity(creature.getId().id()));
}

using namespace de::sots::cellarsandcentaurs::domain::model;

Creature PersistedCreatures::find(const CreatureId& id) {
    auto found = repository_->findById(id.id());
    if (!found) throw NoSuchCreatureException(id);
    return Creature(CreatureId(found->getId()));
}

}
